src/Conversion/ONNXToKrnl/Tensor/Transpose.cpp

```diff
@@ -197,6 +197,7 @@ struct ONNXTransposeOpLowering : public OpConversionPattern<ONNXTransposeOp> {
     create->krnl.iterateIE(loopDef, loopDef, lbs, inUBs,
         [&](KrnlBuilder &createKrnl, ValueRange indices) {
           MultiDialectBuilder<MathBuilder, KrnlBuilder> create(createKrnl);
+          IndexExprScope loopScope(createKrnl);
```

Comment:
Without this, it failed to create `SymbolIndexExpr` because of not the same scope.

Reply:
Sometimes I think the distinction between symbols and dims are really a nuisance, and not sure how key it is anyways... Thought on that?

Reply:
I seems there is no difference if we just do calculation over IndexExpr, but it matters if we get an affine map from IndexExpr. If we have `a*b` in the affine map and `a` is a dim, then `b` must be a literal or symbol. If `b` is symbol, we have a semi affine map.